### PR TITLE
doc: enhance search engine using pymeilisearch 

### LINKED ISSUE #760: Enable PyMeilisearch to improve results in search bar
💡 **Idea**
Let us enable PyMeilisearch in this project so it benefits from all its features.

### PR DESCRIPTION
**Pull request by Revathyvenugopal162**
fix #760

### CODE REVIEW

File: .github/workflows/ci_cd.yml [hunk 1]
```diff
@@ -615,3 +631,31 @@ jobs:
...
+    name: "Deploy stable docs index"
+    runs-on: ubuntu-latest
+    needs: upload_docs_release
+    steps:
+      - name: "Install Git and clone project"
+        uses: actions/checkout@v4
+      - name: "get version of library"
+        shell: bash
+        run: |
+          echo "FULL_VERSION=$(cat VERSION)" >> $GITHUB_ENV
```
Comment (RobPasMue) [inline .github/workflows/ci_cd.yml:645]:
```suggestion
      - name: "Get library version"
        run: |
          echo "FULL_VERSION=$(cat VERSION)" >> $GITHUB_ENV
```
Where is that VERSION coming from @Revathyvenugopal162 ? I haven't defined it anywhere...

File: .github/workflows/ci_cd.yml [hunk 2]
```diff
@@ -615,3 +631,31 @@ jobs:
...
+
+      - name: "Scrape the stable documentation to PyMeilisearch"
+        run: |
+          VERSION=$(python -c "import os; version=os.environ['FULL_VERSION']; print('.'.join(version.split('.')[:2]))")
+          VERSION_MEILI=$(python -c "import os; version=os.environ['FULL_VERSION']; print('-'.join(version.split('.')[:2]))")
+          echo "Calculated VERSION: $VERSION"
+          echo "Calculated VERSION_MEILI: $VERSION_MEILI"
+          echo "VERSION=$VERSION" >> $GITHUB_ENV
+          echo "VERSION_MEILI=$VERSION_MEILI" >> $GITHUB_ENV
```
Comment (RobPasMue) [inline .github/workflows/ci_cd.yml:654]: I think you are missing this in the dev docs index deployment

File: .github/workflows/ci_cd.yml [hunk 3]
```diff
@@ -20,6 +20,8 @@ env:
   RESET_IMAGE_CACHE: 6
   IS_WORKFLOW_RUNNING: True
   ARTIFACTORY_VERSION: v241
+  MEILISEARCH_API_KEY: ${{ secrets.MEILISEARCH_API_KEY }}
+  MEILISEARCH_PUBLIC_API_KEY: ${{ secrets.MEILISEARCH_PUBLIC_API_KEY }}
```
Comment (RobPasMue) [inline .github/workflows/ci_cd.yml:25]: These secrets are at org level, right?

**Review by RobPasMue — CHANGES_REQUESTED** [review]

File: .github/workflows/ci_cd.yml (hunk 2 above)
Comment (Revathyvenugopal162) [reply .github/workflows/ci_cd.yml]: i added it after dev docs deployment.

File: .github/workflows/ci_cd.yml (hunk 3 above)
Comment (Revathyvenugopal162) [reply .github/workflows/ci_cd.yml:25]: yes.

File: .github/workflows/ci_cd.yml (hunk 1 above)
Comment (Revathyvenugopal162) [reply .github/workflows/ci_cd.yml]: solved it.

**Review by RobPasMue — APPROVED** [review]
LGTM! Merging :)

**Pull request merged**